File: mpp/Tema_1_3_V3/src/main/java/Service/ServiceUser.java
package Service;

import Domain.User;
import Repository.UsersRepo;

import java.sql.Connection;

public class ServiceUser implements IServiceUser<User> {

    private UsersRepo ur;

    public ServiceUser(UsersRepo ur) {
        this.ur = ur;
    }

    public ServiceUser() {
        this.ur = new UsersRepo();
    }

    @Override
    public User findByName(String nume) {
        return ur.findByName(nume);
    }

    @Override
    public User findOne(int id) {
        return ur.findOne(id);
    }

    @Override
    public Connection getConnection() {
        return ur.getConnection();
    }
}

File: mpp/LabCSharp_V3/Service/ServiceUser.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using LabCSharp_V3.Repository;
using LabCSharp_V3.Domain;
using System.Data.SQLite;

namespace LabCSharp_V3.Service
{
    class ServiceUser : IServUser
    {

        RepoUser repo;

        public ServiceUser(string name)
        {
            repo = new RepoUser(name);
        }

        public User findByName(string name)
        {
            return repo.findByName(name);
        }
    }
}
